src/ImageSharp/PixelFormats/PixelBlenders/DefaultDarkenPixelBlender{TPixel}.cs

```diff
@@ -0,0 +1,55 @@
+﻿// <copyright file="DefaultDarkenPixelBlender{TPixel}.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.PixelFormats.PixelBlenders
+{
+    using System;
+    using System.Numerics;
+    using ImageSharp.PixelFormats;
+
+    /// <summary>
+    /// Abstract base class for calling pixel composition functions
+    /// </summary>
+    /// <typeparam name="TPixel">The type of the pixel</typeparam>
+    internal class DefaultDarkenPixelBlender<TPixel> : PixelBlender<TPixel>
+        where TPixel : struct, IPixel<TPixel>
+    {
+        /// <summary>
+        /// Gets the static instance of this blender.
+        /// </summary>
+        public static DefaultDarkenPixelBlender<TPixel> Instance { get; } = new DefaultDarkenPixelBlender<TPixel>();
+
+        /// <inheritdoc />
+        public override TPixel Compose(TPixel background, TPixel source, float amount)
+        {
+            return PorterDuffFunctions<TPixel>.DarkenFunction(background, source, amount);
+        }
+
+        /// <inheritdoc />
+        public override void Compose(BufferSpan<TPixel> destination, BufferSpan<TPixel> background, BufferSpan<TPixel> source, BufferSpan<float> amount)
+        {
+            Guard.MustBeGreaterThanOrEqualTo(background.Length, destination.Length, nameof(background.Length));
+            Guard.MustBeGreaterThanOrEqualTo(source.Length, destination.Length, nameof(source.Length));
+            Guard.MustBeGreaterThanOrEqualTo(amount.Length, destination.Length, nameof(amount.Length));
+
+            using (Buffer<Vector4> buffer = new Buffer<Vector4>(destination.Length * 3))
+            {
+                BufferSpan<Vector4> destinationSpan = buffer.Slice(0, destination.Length);
+                BufferSpan<Vector4> backgroundSpan = buffer.Slice(destination.Length, destination.Length);
+                BufferSpan<Vector4> sourceSpan = buffer.Slice(destination.Length * 2, destination.Length);
+
+                PixelOperations<TPixel>.Instance.ToVector4(background, backgroundSpan, destination.Length);
+                PixelOperations<TPixel>.Instance.ToVector4(source, sourceSpan, destination.Length);
+
+                for (int i = 0; i < destination.Length; i++)
+                {
+                    destinationSpan[i] = PorterDuffFunctions.DarkenFunction(backgroundSpan[i], sourceSpan[i], amount[i]);
```

Comment:
actually if we're adding some generation in I'll leave off trying to find an efficient way to wire in the composition modes onto op the blending. At the moment this code is doing the equivalent of 'Over' compositing. But with code generation it will be trivial to expand out the modes to include all the other ones.